includes/class-wp-job-manager-recaptcha.php

```diff
@@ -0,0 +1,236 @@
+		if ( is_null( self::$instance ) ) {
+			self::$instance = new self();
+		}
+		return self::$instance;
+	}
+
+	/**
+	 * Use reCAPTCHA field on the form?
+	 *
+	 * @return bool
+	 */
+	public function use_recaptcha_field() {
+		if ( ! $this->is_recaptcha_available() ) {
+			return false;
+		}
+		return 1 === absint( get_option( 'job_manager_enable_recaptcha_job_submission' ) );
+	}
+
+	/**
+	 * Enqueue the scripts for the form.
+	 */
+	public static function enqueue_scripts() {
+		$instance = self::instance();
+
+		if (
+			$instance->use_recaptcha_field() &&
+			in_array( $instance->recaptcha_version, [ 'v2', 'v3' ], true )
+		) {
+			$recaptcha_version = $instance->recaptcha_version;
+			$recaptcha_url     = '';
+
+			if ( 'v2' === $recaptcha_version ) {
+				$recaptcha_url = 'https://www.google.com/recaptcha/api.js';
+			} elseif ( 'v3' === $recaptcha_version ) {
+				$recaptcha_url = 'https://www.google.com/recaptcha/api.js?render=' . $instance->site_key;
+			}
+			wp_enqueue_script( 'recaptcha', $recaptcha_url, [], JOB_MANAGER_VERSION, [ 'strategy' => 'defer' ] );
+		}
+	}
+
+	/**
+	 * Checks whether reCAPTCHA has been set up and is available.
+	 *
+	 * @return bool
+	 */
+	public function is_recaptcha_available() {
+		$is_recaptcha_available = ! empty( $this->site_key ) && ! empty( $this->secret_key );
+
+		/**
+		 * Filter whether reCAPTCHA should be available for this form.
+		 *
+		 * @since 1.30.0
+		 *
+		 * @param bool $is_recaptcha_available
+		 */
+		return apply_filters( 'job_manager_is_recaptcha_available', $is_recaptcha_available );
+	}
+
+	/**
+	 * Dispaly the reCAPTCHA field in the form.
+	 *
+	 * @return void
+	 */
+	public function display_recaptcha_field() {
+		$field             = [];
+		$field['label']    = get_option( 'job_manager_recaptcha_label' );
+		$field['required'] = true;
+		$field['site_key'] = $this->site_key;
+
+		$template = 'form-fields/recaptcha-' . ( 'v3' === $this->recaptcha_version ? 'v3-' : '' ) . 'field.php';
+
+		get_job_manager_template(
+			$template,
+			[
+				'key'   => 'recaptcha',
+				'field' => $field,
+			]
+		);
+	}
+
+	/**
+	 * Validate a reCAPTCHA field.
+	 *
+	 * @param bool $success
+	 *
+	 * @return bool|WP_Error
+	 */
+	public function validate_recaptcha_field( $success ) {
+		$recaptcha_field_label = get_option( 'job_manager_recaptcha_label' );
+
+		// translators: %s is the name of the form validation that failed.
+		$validation_error = new WP_Error( 'validation-error', sprintf( esc_html__( '"%s" check failed. Please try again.', 'wp-job-manager' ), $recaptcha_field_label ) );
+
+		// phpcs:ignore WordPress.Security.NonceVerification.Missing -- Nonce check happens earlier (when possible).
+		$input_recaptcha_response = isset( $_POST['g-recaptcha-response'] ) ? sanitize_text_field( wp_unslash( $_POST['g-recaptcha-response'] ) ) : '';
+
+		if ( empty( $input_recaptcha_response ) ) {
+			return $validation_error;
+		}
+
+		if ( 'v2' === $this->recaptcha_version ) {
```

Comment:
Something that you can consider as an improvement. This seems like a good case of defining a base class for recaptcha and then create two subclasses for v2 and v3. You might need to define some extra methods that are overridden in subclasses. This way we can avoid the separate flows for each version.

Reply:
Thanks @gikaragia !  I had a talk with @fjorgemota  about this and after discussing this and a few other approaches we thought maybe since it's just small we should keep it as-is for now.  Appreciate the suggestion though!!